pkg/controllers/image/image.go

```diff
@@ -278,6 +279,32 @@ func (h handler) onChangeGitRepo(gitrepo *v1alpha1.GitRepo, status v1alpha1.GitR
 	return status, err
 }
 
+func (h handler) setupKnownHosts(gitrepo *v1alpha1.GitRepo, data []byte) error {
+	tmpdir, err := os.MkdirTemp("", gitrepo.Namespace+"-"+gitrepo.Name)
```

Comment:
I wouldn't use user input here to construct the temp dir, should be safe and a [random suffix](https://pkg.go.dev/os#MkdirTemp) is added, but still.

```suggestion
	tmpdir, err := os.MkdirTemp("", "fleet-")
```

Reply:
Is there any chance to have some sort of collusion here? Like two different process fetching the digest for the same tag at a slightly different type which could lead to different digest for the same tag?